### PR TITLE
Migrate remaining 'block switcher' e2e tests to Playwright

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
Part of #38851.

PR migrates `block-switcher.test.js` e2e tests to Playwright.

## Why?
The test has been failing a lot recently. Ref: #40369.

## Testing Instructions
```
npm run test:e2e:playwright -- test/e2e/specs/editor/various/block-switcher.spec.js
```

### CODE REVIEW

File: test/e2e/specs/editor/various/block-switcher.spec.js [hunk 1]
```diff
@@ -8,6 +8,179 @@ test.describe( 'Block Switcher', () => {
...
+		await editor.canvas
+			.getByRole( 'button', { name: 'Add default block' } )
+			.click();
+		await page.keyboard.type( '- List content' );
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '## I am a header' );
+		await page.keyboard.down( 'Shift' );
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.up( 'Shift' );
```
Comment (Mamaduka) [inline test/e2e/specs/editor/various/block-switcher.spec.js:134]: Note for future me: If this fails again, ensure the block toolbar is displayed by adding `await pageUtils.pressKeys( 'alt+F10' );`.

Currently, I can reproduce the scenario with Playwright, but I can do it manually sometimes.

File: packages/e2e-tests/specs/editor/various/block-switcher.test.js [hunk 2]
```diff
@@ -1,130 +0,0 @@
...
-	insertBlock,
-	pressKeyWithModifier,
-} from '@wordpress/e2e-test-utils';
-
-describe( 'Block Switcher', () => {
-	beforeEach( async () => {
-		await createNewPost();
-	} );
-
-	it( 'Should show the expected block transforms on the list block when the blocks are removed', async () => {
```
Comment (t-hamano) [inline packages/e2e-tests/specs/editor/various/block-switcher.test.js:17]: How about making "Should" a lower case along with other tests?

File: test/e2e/specs/editor/various/block-switcher.spec.js [hunk 3]
```diff
@@ -8,6 +8,179 @@ test.describe( 'Block Switcher', () => {
...
+			.click();
+
+		await expect(
+			page
+				.getByRole( 'menu', { name: 'Multiple blocks selected' } )
+				.getByRole( 'menuitem', { name: 'Columns' } )
+		).toBeHidden();
+	} );
+
 	test( 'Block variation transforms', async ( { editor, page } ) => {
```
Comment (t-hamano) [inline test/e2e/specs/editor/various/block-switcher.spec.js]: This is a test that already existed, but how about adjusting the title to match the other tests? For example, "should be able to transform to block variations."?

File: test/e2e/specs/editor/various/block-switcher.spec.js [hunk 4]
```diff
@@ -8,6 +8,179 @@ test.describe( 'Block Switcher', () => {
...
 		// This is the `stack` Group variation.
```
Comment (t-hamano) [inline test/e2e/specs/editor/various/block-switcher.spec.js:188]:
```suggestion
		// This is the `flex` Group variation.
```

This looks like a typo.

**Review by t-hamano — APPROVED** [review]
LGTM 👍

Although I suggested a small change, I ran this test in debug mode and confirmed that the test worked correctly.

BTW I learned that this test migration was also attempted in #50845. How about adding him as a co-author?

File: packages/e2e-tests/specs/editor/various/block-switcher.test.js (hunk 2 above)
Comment (Mamaduka) [reply packages/e2e-tests/specs/editor/various/block-switcher.test.js:17]: I think it's okay. We don't really have a strict guideline when it comes to naming.

File: test/e2e/specs/editor/various/block-switcher.spec.js (hunk 4 above)
Comment (Mamaduka) [reply test/e2e/specs/editor/various/block-switcher.spec.js:188]: I don't think so; the variation name is `stack`. The flex is just implementation details.

**Pull request merged**